marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.tsx

```diff
@@ -47,51 +57,74 @@ const EntityTypeModal: React.FC<Props> = (props) => {
     if (event.target.id === 'description') {
       setDescription(event.target.value);
     }
-  }
+  };
 
   const onOk = (event) => {
     event.preventDefault();
-    if (!NAME_REGEX.test(name)) {
-      setErrorMessage(ModelingTooltips.addEntityName)
+    if (props.isEditModal) {
+      updateEntityDescription(name, description);
     } else {
-      createEntityType(name, description);
+      if (!NAME_REGEX.test(name)) {
+        setErrorMessage(ModelingTooltips.addEntityName)
+      } else {
+        createEntityType(name, description);
+      }
+    }
+  };
+
+  const updateEntityDescription = async (name: string, description: string) => {
+    try {
+      const response = await updateModelInfo(name, description);
+      if (response['status'] === 200) {
+        props.updateEntityTypesAndHideModal();
+      }
+    } catch (error) {
+      if (error.response.status === 400) {
```

Comment:
I keep seeing this code repeated in the UI - couldn't this be in some sort of "common" library and reused, perhaps as a callback/promise so that if hasOwnProperty('message') is true, then the caller determines what to do, with the common function providing the value of error['response']['data']['message']?

Reply:
Good point.
@rjrudin, @brucean52, what do you think about this?
```
  const sendAndHandle = (requestFunction, errorFunction) => {
    try {
      requestFunction();
    } catch (error) {
      if (error.response.status === 400) {
        if (error.response.data.hasOwnProperty('message')) {
          errorFunction(error['response']['data']['message']);
        }
      } else {
        handleError(error);
      }
    } finally {
      resetSessionTime();
    }
  };
```
We can then use the requestFunction to make the request, open/close modal, etc. Similary we can also use errorFunction to set error message, open/close modal, etc.

Reply:
I think this could work, but make the errorFunction conditional more generic to support more customization

Reply:
After a couple of discussions, @brucean52  and I came down with the following function:
```
function sendAndHandle(requestFunction, errorFunction, resetSessionTime) {
  try {
    requestFunction();
  } catch (error) {
    errorFunction(error);
  } finally {
    resetSessionTime();
  }
}
export default sendAndHandle;
```
The entire catch block is replaced with `errorFunction(error);` in order to support more customization and we need to pass in the `resetSessionTime` function since `useContext` hook for `resetSessionTime` only works in react components and this isn't one.

At this point, this just seems to be a try/catch handler and it doesn't seem worth adding it to a common library.

Reply:
This seems like it could be part of that facade library we've discussed - i.e. where ideally, every HTTP call goes through a very narrow chunk of code that can handle generic concerns for every HTTP call.